storey/aggregations.py

```diff
@@ -638,17 +693,20 @@ def initialize_from_data(self, data, base_time):
             timestamp1, timestamp2 = aggregation_bucket_initial_data.keys()
             first_time, last_time = min(timestamp1, timestamp2), max(timestamp1, timestamp2)
 
-        bucket_index = self.window.total_number_of_buckets - 1
-        self.last_bucket_start_time = self.window.get_window_start_time_by_time(base_time)
+        bucket_index = total_buckets - 1
+        if self.explicit_windows:
```

Comment:
You are doing a lot of these conditions maybe just check the condition once on init and save the desired `get_window_start_time_by_time` function in `self` and just call it every time